apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/AbstractSpan.java

```diff
@@ -584,4 +617,39 @@ public LongList getChildIds() {
     }
 
     protected abstract T thiz();
+
+    /**
+     * @return user outcome if set, otherwise outcome value
+     */
+    public Outcome getOutcome() {
+        if (userOutcome != null) {
+            return userOutcome;
+        }
+        return outcome == null ? Outcome.UNKNOWN : outcome;
+    }
+
+    /**
+     * Sets outcome
+     *
+     * @param outcome outcome
+     * @return this
+     */
+    public T withOutcome(Outcome outcome) {
```

Comment:
this will be the method called by automatic instrumentation, thus it should only be set once, the first value set wins.